tests/data_group_test.py

```diff
@@ -252,12 +252,10 @@ def test_getitem_positional_indexing_without_dim_label_raises_unless_1d():
         dg2d[::]
 
 
-def test_getitem_positional_indexing_raises_when_length_is_not_unique():
+def test_getitem_positional_indexing_does_not_raise_when_length_is_not_unique():
     dg = sc.DataGroup({'a': sc.arange('x', 4), 'b': sc.arange('x', 5)})
-    with pytest.raises(sc.DimensionError):
-        dg['x', 2]
-    with pytest.raises(sc.DimensionError):
-        dg['x', 2:3]
+    dg['x', 2]
```

Comment:
I think this one must still raise! Try it on `list` or NumPy array.

Reply:
There may be another option: Instead of raising, drop items that fall out of range. But that would imply returning an empty data group in some cases, so I am not sure this is the optimal choice.

Reply:
Which one must still raise? `dg['x', 2]` or `dg['x', 2:3]` or both?

Reply:
> Which one must still raise? `dg['x', 2]` or `dg['x', 2:3]` or both?

The former (the line I commented on).

Reply:
> > There may be another option: Instead of raising, drop items that fall out of range. But that would imply returning an empty data group in some cases, so I am not sure this is the optimal choice.
> 
> Why drop the empty items? It's legal to have empty items in a datagroup. I'd probably be more surprised if my item disappeared than it being length 0 after slicing.

Because `dg['x', i]` must return something that does *not* have the `'x'` dimension. Therefore it cannot be length-0.

Reply:
Just to be clear, in this example:

```python
dg = sc.DataGroup({'a': sc.arange('x', 4), 'b': sc.arange('x', 5)})
dg['x', 2]
```
Do you mean that the last line should raise an exception?

Reply:
Well, depends on what I wrote above:

> There may be another option: Instead of raising, drop items that fall out of range. But that would imply returning an empty data group in some cases, so I am not sure this is the optimal choice.

I think the options for positional integer-indexing (not range) are:

1. Always raise if length is `None`.
2. Raise if at least one item is out of bounds.
3. Never raise, drop out-of-bounds items.

I am not certain what the optimal choice is here?

Reply:
I think it makes sense to keep the logic simple with few extra rules, for example:
  * Indexing or slicing a dg indexes/slices all items that have the relevant dimension.
  * If one of them raises when indexed/sliced then then indexing/slicing all of them also raises.

Reply:
I agree with the last comment. This is generally how we map operations over data groups. E.g.
```python
dg = sc.DataGroup({
    'a': sc.arange('x', 4),
    'b': sc.arange('y', 5)
})
dg.sum('x')
```
raises because `'b'` doesn't have an `'x'`-dim. I.e., a failure in one items makes the whole operation fail.

Reply:
👍 Agree, go ahead with this solution then.